src/main/java/com/global/api/gateways/Gateway.java

```diff
@@ -153,4 +169,12 @@ private String buildQueryString(HashMap<String, String> queryStringParams) throw
         }
         return sb.toString();
     }
+    
+    private String parseErrorMessage(String rawResponse) throws ApiException {
```

Comment:
This is specifically looking for XML to be returned and this is not the case with all of the gateways inheriting from this class. This should be made abstract and the XML implementation moved to the XmlGateway.java Class.

Reply:
This one is simple enough.

Change this method signature to:
protected abstract String parseErrorMessage(String rawResponse) throws ApiException;

Take the existing code you have here and move it to the XmlGateway.java class as the implementation of the abstract method.